### ISSUE TITLE
ChatGPT Sentiment Analysis

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#411]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : ChatGPT Sentiment Analysis</br>
:red_circle: **Aim** : The aim of this project is to analyze the sentiments of the tweets made on/against ChatGPT.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/charunisa/chatgpt-sentiment-analysis</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by JIGYASAKARAKOTI** [comment]
Full name : Jigyasa Karakoti
GitHub Profile Link: https://github.com/JIGYASAKARAKOTI
Email ID: jigyasakarakoti@gmail.com
Participant ID (if applicable):
Approach for this Project: I'll be using https://www.kaggle.com/datasets/charunisa/chatgpt-sentiment-analysis/data  this dataset and first do some EDA then will go for data cleaning and preprocessing after that feature engineering
Convert the tokenized text data into numerical vectors using TF-IDF, and Word Embeddings like Word2Vec or GloVe, will split the dataset into training and testing sets. Going to Implement 3 to 4 different sentiment analysis algorithms like Naive Bayes, Support Vector Machines (SVM), Random Forest, and Deep Learning models like LSTM or GRU. If the results are satisfactory then we will compare the results by using the R-squared metric. If applicable, we will perform hyperparameter tuning on the best-performing model to optimize its performance. Will update you if I find something new in it.
What is your participant role? social winter of code 2024.

**Comment by abhisheks008** [comment]
Hi @JIGYASAKARAKOTI it's a nice approach. Try to majorly focus on DL model implementations. Issue assigned to you. You can start working on it, all the best!

**Comment by JIGYASAKARAKOTI** [comment]
Thank you so much, till when i can submit the solution? @abhisheks008

**Comment by abhisheks008** [comment]
> Thank you so much, till when i can submit the solution? @abhisheks008

You can take your time, no issues with that.

**Comment by JIGYASAKARAKOTI** [comment]
@abhisheks008 sir do i have to make a new project on jupyter or i have to make changes on  given code in readme.

**Comment by abhisheks008** [comment]
I didn't get you.
@JIGYASAKARAKOTI

**Comment by JIGYASAKARAKOTI** [comment]
i have made the entire project into a new folder. @abhisheks008

**Comment by abhisheks008** [comment]
You have created your project in the local system right? First fork this repository, upload your project in the forked repo, push the code by creating a pull request.

**Comment by JIGYASAKARAKOTI** [comment]
Sir dataset is greater than 25 mb, I'm not able to upload it.

**Comment by JIGYASAKARAKOTI** [comment]
sir , the project is in a folder in my system . how can i upload the entire folder, and if im doing it dataset is of 33MB which is greater than 25 MB So i 'm unable to upload it. @abhisheks008

**Comment by abhisheks008** [comment]
If your dataset is more than the threshold, then upload the URL in a README file inside the Dataset folder. That'll do the same job.

**Comment by dipayan22** [comment]
Full name : Dipayan Majumder
GitHub Profile Link : https://github.com/dipayan22
Email ID : majumderd007@gmail.com
Participant ID (if applicable): 
Approach for this Project : Clearly outline project goals and specify the type of text data for analysis . Assemble a labeled dataset and preprocess the text (cleaning, tokenization) . Select a suitable model, either rule-based or machine learning-based (e.g., Naive Bayes, SVM) or DL model like RNN . Split the dataset, train the chosen model, and fine-tune for optimal performance.
Assess the model's accuracy, precision, recall, and F1-score on a testing dataset.
Integrate the trained model into the application or system. 

What is your participant role? (Mention the Open Source program) SWOC 2014

**Comment by abhisheks008** [comment]
> Full name : Dipayan Majumder GitHub Profile Link : https://github.com/dipayan22 Email ID : [majumderd007@gmail.com](mailto:majumderd007@gmail.com) Participant ID (if applicable): Approach for this Project : Clearly outline project goals and specify the type of text data for analysis . Assemble a labeled dataset and preprocess the text (cleaning, tokenization) . Select a suitable model, either rule-based or machine learning-based (e.g., Naive Bayes, SVM) or DL model like RNN . Split the dataset, train the chosen model, and fine-tune for optimal performance. Assess the model's accuracy, precision, recall, and F1-score on a testing dataset. Integrate the trained model into the application or system.
> 
> What is your participant role? (Mention the Open Source program) SWOC 2014

This issue is already assigned to a participant.
